src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -273,6 +272,13 @@ export async function uploadPayload(payloadFileName: string) {
         // Pass along error to callee function
         throw new ToolkitError(errorMessage, { cause: e as Error })
     }
+    try {
+        await uploadArtifactToS3(payloadFileName, response)
+    } catch (e: any) {
+        const errorMessage = e?.message || 'Error in uploadArtifactToS3 call'
+        throw new Error(errorMessage)
+    }
+    return response.uploadId
```

Comment:
Moved this try-catch down here so that error handling is separate for the CreateUploadUrl and uploadArtifactToS3, rather than bundling it all in one try block which doesn't tell us exactly which method errored out.